src/components/badge/badge.test.js

```diff
@@ -0,0 +1,87 @@
+import * as React from "react";
+import Badge from "./badge.component";
+import Button from "../button/button.component";
+import CypressMountWithProviders from "../../../cypress/support/component-helper/cypress-mount";
+import {
+  badge,
+  badgeCounter,
+  badgeCrossIcon,
+} from "../../../cypress/locators/badge";
+
+const BadgeComponent = ({ ...props }) => {
+  return (
+    <Badge {...props}>
+      <Button
+        style={{
+          marginRight: 0,
+        }}
+        buttonType="tertiary"
+      >
+        Filter
+      </Button>
+    </Badge>
+  );
+};
+
+context("Testing Badge component", () => {
+  describe("should render Badge component", () => {
+    it.each([[1], [99]])("should check Badge counter is set to %s", (value) => {
+      CypressMountWithProviders(<BadgeComponent counter={value} />);
+
+      badgeCounter()
+        .invoke("show")
+        .should("be.visible")
+        .invoke("text")
+        .and("contain", value);
+    });
+
+    it.each([[100], [999]])(
+      "should check Badge counter is set to 99 using %s as input",
+      (value) => {
+        CypressMountWithProviders(<BadgeComponent counter={value} />);
+
+        badgeCounter()
+          .invoke("show")
+          .should("be.visible")
+          .invoke("text")
+          .and("contain", 99);
+      }
+    );
+
```

Comment:
Question on this please Stephen. The purpose of this test is to mount the badge with the counter value of 999 and to verify the counter is actually set to 99 as that's the  max value allowed. If there was a bug and the counter was actually set to 999, this test would still pass, wouldn't it? Because 999 contains 99. Could you mount it with a value of 980 so the verification would fail when it should? Or if there is a reason to use 999, could we use "equals" instead of "contain" for the verification? Thanks.

Reply:
Okay, I've made that change.

Reply:
Thanks Stephen

Reply:
or it could be also changed to `equals` and the TC using `999` could stay unchanged.